components/HttpsCoponent_/HttpsRequest_.c

```diff
@@ -0,0 +1,315 @@
+    ESP_LOGI(TAG, "https_request using global ca_store");
+    esp_ret = esp_tls_set_global_ca_store(server_root_cert_pem_start, server_root_cert_pem_end - server_root_cert_pem_start);
+    if (esp_ret != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Error in setting the global ca store: [%02X] (%s),could not complete the https_request using global_ca_store", esp_ret, esp_err_to_name(esp_ret));
+        return;
+    }
+    esp_tls_cfg_t cfg = {
+        .use_global_ca_store = true,
+    };
+    https_get_request(cfg, Web_URL, HttpsBuf);
+    esp_tls_free_global_ca_store();
+}
+
+#ifdef CONFIG_EXAMPLE_CLIENT_SESSION_TICKETS
+static void https_get_request_to_local_server(const char *url)
+{
+    ESP_LOGI(TAG, "https_request to local server");
+    esp_tls_cfg_t cfg = {
+        .cacert_buf = (const unsigned char *)local_server_cert_pem_start,
+        .cacert_bytes = local_server_cert_pem_end - local_server_cert_pem_start,
+        .skip_common_name = true,
+    };
+    save_client_session = true;
+    https_get_request(cfg, url, LOCAL_SRV_REQUEST);
+}
+
+static void https_get_request_using_already_saved_session(const char *url)
+{
+    ESP_LOGI(TAG, "https_request using saved client session");
+    esp_tls_cfg_t cfg = {
+        .client_session = tls_client_session,
+    };
+    https_get_request(cfg, url, LOCAL_SRV_REQUEST);
+    esp_tls_free_client_session(tls_client_session);
+    save_client_session = false;
+    tls_client_session = NULL;
+}
+#endif
+
+static void https_request_task(void *pvparameters)
+{
+    ESP_LOGI(TAG, "Start https_request example");
+
+#ifdef CONFIG_EXAMPLE_CLIENT_SESSION_TICKETS
+    char *server_url = NULL;
+#ifdef CONFIG_EXAMPLE_LOCAL_SERVER_URL_FROM_STDIN
+    char url_buf[SERVER_URL_MAX_SZ];
+    if (strcmp(CONFIG_EXAMPLE_LOCAL_SERVER_URL, "FROM_STDIN") == 0)
+    {
+        example_configure_stdin_stdout();
+        fgets(url_buf, SERVER_URL_MAX_SZ, stdin);
+        int len = strlen(url_buf);
+        url_buf[len - 1] = '\0';
+        server_url = url_buf;
+    }
+    else
+    {
+        ESP_LOGE(TAG, "Configuration mismatch: invalid url for local server");
+        abort();
+    }
+    printf("\nServer URL obtained is %s\n", url_buf);
+#else
+    server_url = CONFIG_EXAMPLE_LOCAL_SERVER_URL;
+#endif /* CONFIG_EXAMPLE_LOCAL_SERVER_URL_FROM_STDIN */
+    https_get_request_to_local_server(server_url);
+    https_get_request_using_already_saved_session(server_url);
+#endif
+
+#if CONFIG_MBEDTLS_CERTIFICATE_BUNDLE
+    https_get_request_using_crt_bundle();
+#endif
+    ESP_LOGI(TAG, "Minimum free heap size: %" PRIu32 " bytes", esp_get_minimum_free_heap_size());
+    https_get_request_using_cacert_buf();
+    https_get_request_using_global_ca_store();
+    ESP_LOGI(TAG, "Finish https_request example");
+    free(HttpsBuf);
+    free(Web_URL);
+    free(WebServerAddress);
+    vTaskDelete(NULL);
+}
+/**
+
+*@brief This function performs an HTTPS GET request to a specified server URL with the provided header request.
```

Comment:
this function is public, so define these help lines in .h files

Reply:
ok